src/Storages/Hive/StorageHive.cpp

```diff
@@ -542,7 +543,34 @@ HiveFilePtr StorageHive::createHiveFileIfNeeded(
     }
     return hive_file;
 }
+bool StorageHive::isColumnOriented()
+{
+    return format_name == "Parquet" || format_name == "ORC";
+}
 
+Block StorageHive::getActualColumnsToRead(Block sample_block, const Block & header_block, const NameSet & partition_columns)
+{
+    if (!isColumnOriented())
+        return header_block;
+    Block result_block = sample_block;
+    for (const auto & column : partition_columns)
```

Comment:
In this case it seems possible that `result_block` does not contain requested columns (`column_names`)? It will not work correctly in this case

Reply:
the `sample_block` contains all of the columns from `column_names`
```
    const auto & header_block = metadata_snapshot->getSampleBlock();
    Block sample_block;
    for (const auto & column : column_names)
    {
        sample_block.insert(header_block.getByName(column));
        if (column == "_path")
            sources_info->need_path_column = true;
        if (column == "_file")
            sources_info->need_file_column = true;
    }

    sample_block = getActualColumnsToRead(sample_block, header_block, NameSet{partition_names.begin(), partition_names.end()});
```

the `result_block` could contain more columns then `sample_block` in the case that all of the requried columns are partition columns, but never less.